wpilibNewCommands/src/test/native/cpp/frc2/command/CommandDecoratorTest.cpp

```diff
@@ -152,3 +153,47 @@ TEST_F(CommandDecoratorTest, Unless) {
   scheduler.Run();
   EXPECT_TRUE(hasRun);
 }
+
+TEST_F(CommandDecoratorTest, FinallyDo) {
+  CommandScheduler scheduler = GetScheduler();
+  int counter = 0;
+  CommandPtr command = FunctionalCommand([] {}, [] {},
+                                         [&counter](bool interrupted) {
+                                           if (!interrupted) {
+                                             counter++;
+                                           }
+                                         },
+                                         [] { return true; })
+                           .FinallyDo([&counter](bool interrupted) {
+                             if (!interrupted) {
+                               counter *= 2;
```

Comment:
`*= 2` when you start from 0 doesn’t seem like a good test? Did you mean `+= 2`?

Reply:
Nope, I wrote what I meant.
That detail is to ensure that `counter++` (`end()`) runs before decorator-injected lambda.

Reply:
Two booleans doesn't do anything about execution order, which is what I want to check.

Reply:
if you check the state of one boolean when setting the other, it should be able to work, I think?

Reply:
Example?

Reply:
if (prevFlag == true) orderFlag = true